jsk_apc2016_common/samples/sample-fetch-pick.l

```diff
@@ -24,54 +39,284 @@
   (move 6000)
-  (test2 (+ x 50) y z) ; move end-coords x +250
+  
+  (send *fetch* :inverse-kinematics (make-coords :pos (float-vector x y (+ z 200))) :debug-view t) ;; move hand to over the object (nearer than the previous line)
   (move 3000)
-  (try-grasp)
-  ;(test2 (+ x 50) y (+ z 100)) ; move end-coords z +100
-  ;(move 3000)
+  
+  ;(send *fetch* :inverse-kinematics (make-coords :pos (float-vector x y z) :rpy (float-vector 0 (/ pi 4) 0)) :debug-view t)
+  (send *fetch* :inverse-kinematics (make-coords :pos (float-vector x y z)) :debug-view t) ;; move hand to the very place where the target object exists
+  (move 3000))
+  
+  
+(defun try-grasp()
+  (send *ri* :go-grasp :effort 60))
+
+
+;;main function.
+;;try to grasp object.
+(defun trial1()
+  (send *ri* :angle-vector (send *fetch* :reset-pose))
+  (send *ri* :stop-grasp)
+  (send *ri* :wait-interpolation)
+  (format t "debug 1~%")
+
+  (if (search-for-object)
+      (progn
+	(format t "debug 2~%")
+	(if recon-table-edge
+	    (move-with-certain-path))
+	(format t "debug 3~%")
+	(try-grasp))
+    (format t "trial canceled.~%"))
+  )
+
+(defun trial2()
+  (send *ri* :angle-vector (send *fetch* :reset-pose))
+  (send *ri* :stop-grasp)
+  (send *ri* :wait-interpolation)
+  (format t "debug 1~%")
+
+  (if (search-for-object)
+      (progn
+	(format t "debug 2~%")
+	(if recon-table
+	    (progn
+	      (make-table table-x table-y 0)
+	      (send *fetch* :inverse-kinematics (make-coords :pos (float-vector x y z)) :debug-view t)
+	      (move 10000)))
+	(format t "debug 3~%")
+	(try-grasp))
+    (format t "trial canceled.~%"))
+  )
+
+(defun trial3()
+  (send *ri* :angle-vector (send *fetch* :reset-pose))
+  (send *ri* :stop-grasp)
+  (send *ri* :wait-interpolation)
+
+  ;;pick object
+  (if (search-for-object)
+      (progn
+	;;extend torso in order to expand view
+	;(send *fetch* :torso :waist-z :joint-angle 300)
+	;(move 3000)
+	;(look-at)
+	(ros::publish "stop_moveit" msg)
+	(send *fetch* :inverse-kinematics (make-coords :pos (float-vector (- x 200) y (+ z 200))) :debug-view t)
+	(move 10000)
+	(let ((flag nil) (i 0))
+	  (while (null flag)
+	    (send *fetch* :inverse-kinematics (make-coords :pos (float-vector x y (+ z i))))
+	    (setq flag (move 5000 1))
+	    (setq i (+ i 5))))
+	(send *ri* :wait-interpolation)
+	
+	(try-grasp)
+	
+	(send *fetch* :inverse-kinematics (make-coords :pos (float-vector (- x 200) y (+ z 200))))
+	(move 1000)
+	(send *fetch* :inverse-kinematics (make-coords :pos (float-vector (- x 300) y (+ z 100))))
+	(move 2000)
+	(send *fetch* :inverse-kinematics (make-coords :pos (float-vector 350 600 (+ z 100))) :debug-view t)
+	(move 5000)
+	(send *fetch* :inverse-kinematics (make-coords :pos (float-vector 350 600 100)) :debug-view t :rotation-axis nil)
+	(move 5000)
+	(send *ri* :stop-grasp)
+	(init))
+    (format t "trial canceled.~%"))
```

Comment:
@YutoUchimi
 I don't force you to remove `traial1` and `trial2`. 
If you want to leave them, you can rename them, for example `simple-pick` or  picking-without-collision-detect`.
If you really do not need the functions, you can remove them.